datasource/src/main/scala/quasar/physical/s3/impl/evaluate.scala

```diff
@@ -76,59 +87,4 @@ object evaluate {
           case s => Sync[F].raiseError(new Exception(s"Unexpected status $s"))
         }
     }
-  }
-
-  private def noParseError(path: AFile, parsing: S3JsonParsing, message: String): ResourceError = {
-    val noParseMsg = s"Could not parse the file as JSON. Ensure you've configured the correct jsonParsing option for this bucket: $message"
-    val expectedFormat = parsing match {
-      case S3JsonParsing.LineDelimited => "Newline-delimited JSON"
-      case S3JsonParsing.JsonArray => "Array-wrapped JSON"
-    }
-
-    ResourceError.malformedResource(
-      ResourcePath.Leaf(path),
-      expectedFormat,
-      noParseMsg)
-  }
-
-  // putting it all together.
-  def apply[F[_]: Effect: Timer](
-    jsonParsing: S3JsonParsing,
-    client: Client[F],
-    uri: Uri,
-    file: AFile,
-    sign: Request[F] => F[Request[F]])
-    (implicit MR: MonadResourceErr[F])
-      : F[Option[Stream[F, Data]]] = {
-    // convert the pathy Path to a POSIX path, dropping
-    // the first slash, like S3 expects for object paths
-    val objectPath = Path.posixCodec.printPath(file).drop(1)
-
-    // Put the object path after the bucket URI
-    val queryUri = appendPathUnencoded(uri, objectPath)
-    val request = Request[F](uri = queryUri)
-
-    // figure out how we're going to parse the object as JSON
-    val circeJsonPipe = circePipe[F](jsonParsing)
-
-    sign(request) >>= { r =>
-      val stream =
-        OptionT(streamRequestThroughFs2[F, Data](client, r) { resp =>
-          // convert the data to JSON, using the parsing method
-          // of our choice
-          val asJson: Stream[F, Json] = resp.body.through(circeJsonPipe)
-
-          // convert the JSON from circe's representation to ours
-          val asData: Stream[F, Data] = asJson.map(circeJsonToData)
-
-          // and we're done.
-          asData
-        })
-
-      stream.map(_.handleErrorWith {
```

Comment:
We might want to still handle errors like this. Let me see what `jawnfs2` does with errors.

Reply:
Reinstated it.